tests/test_config.py

```diff
@@ -0,0 +1,52 @@
+#! /usr/bin/env python
+# vim: expandtab shiftwidth=4 softtabstop=4 tabstop=17 filetype=python :
+import argparse
+import os
+import sys
+import unittest
+
+from caramel import config
+from tests import fixtures
+
+
+class TestConfig(unittest.TestCase):
+    def test_inifile_argument(self):
+        my_ini = "/a/path/to/my_ini_file.ini"
+
+        parser = argparse.ArgumentParser()
+        config.add_inifile_argument(parser)
+        args = parser.parse_args([my_ini])
+
+        self.assertEqual(args.inifile, my_ini)
+        return True
+
+    def test_inifile_env(self):
+        my_ini = "/a/path/to/my_ini_file.ini"
+
+        os.environ["CARAMEL_INI"] = my_ini
+
+        parser = argparse.ArgumentParser()
+        config.add_inifile_argument(parser)
+        args = parser.parse_args([])
+
+        self.assertEqual(args.inifile, my_ini)
+        return True
+
+    def test_inifile_argument_env(self):
+        my_ini = "/a/path/to/my_ini_file.ini"
+
+        other_ini = "/a/path/to/other_ini_file.ini"
+
+        sys.argv.append(my_ini)
+        os.environ["CARAMEL_INI"] = other_ini
+
+        parser = argparse.ArgumentParser()
+        config.add_inifile_argument(parser)
+        args = parser.parse_args([my_ini])
+
+        self.assertEqual(args.inifile, my_ini)
+        return True
+
+    def test_no_ini_path(self):
+        with self.assertRaises(ValueError):
```

Comment:
No need to use a fixture here for a single test-case, that only obscures what the test-case does.

Reply:
I understand that, but I've locally continued development on adding loggers and their test uses an extended version of this fixture(in my sketches), so is there really a need to change this here when it will be extended and used in the near future(<1 month)?

Reply:
Premature abstractions and abstractions with a single user are generally to be avoided, especially when spread out across different files.

Overall, what you want to optimize for is making the single case easy to understand. You've already hit the impact of abstract difficulty there was to work with the fixtures in the model test-cases, and those are shared between  view and model tests.

So, since this (config parsing) doesn't involve any shared state, the fixture and abstractions should be as close as possible to the tests, in this case, preferrably in the same case.

Just as you currently have no use for inheriting "SImilarlyComparable" as the tests don't actually use that.

Reply:
Nvm, Action-based implementation of the check for an ini-file made it redundant in this patch.